### PR TITLE
Allow finer-grained customization of the status endpoint

### LINKED ISSUE #4737: Status endpoint for Reader Cromwells
WAAS will have different `/status` requirements than the reader or writer Cromwells.

This could involve (depending on time available) either:
* Generalizing the current status implementation to allow callers of `/status` to opt-in/opt-out of status information they do or don't care about
* Creating a new specialist health implementation which knows what WAAS would care about.

### PR DESCRIPTION
**Pull request by cjllanwarne**
- [x] Necessitates an update to https://github.com/broadinstitute/firecloud-develop/pull/1598
- Closes #4737

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,78 @@
...
+#### HealthMonitor Configuration
+
+The HealthMonitor configuration has been refactored to provide a simpler interface:
+* You no longer need to specify a monitor class in your `cromwell.conf` as this will now be inherited from the `reference.conf` value.
+* You can now opt-in and opt-out of any combination of status monitors.
+* The PAPI backends to monitor can now be listed in a single field.
+
+##### Upgrading
+
+You are now able to put any combination of health-checks that you desire in your configuration. However to upgrade directly
```
Comment (aednichols) [inline CHANGELOG.md]: [ToL] "You can now put any combination of health checks in your configuration"?

File: services/src/main/scala/cromwell/services/healthmonitor/impl/ChooseYourOwnAdventureHealthMonitorServiceActor.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package cromwell.services.healthmonitor.impl
+
+import akka.actor.ActorRef
+import com.typesafe.config.Config
+import cromwell.services.healthmonitor.HealthMonitorServiceActor
+import cromwell.services.healthmonitor.impl.workbench.WorkbenchHealthMonitorServiceActor
+
+final class ChooseYourOwnAdventureHealthMonitorServiceActor(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef)
```
Comment (aednichols) [inline services/src/main/scala/cromwell/services/healthmonitor/impl/ChooseYourOwnAdventureHealthMonitorServiceActor.scala]: Hmm

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,78 @@
...
+services {
+    HealthMonitor {
+        class = "cromwell.services.healthmonitor.impl.workbench.WorkbenchHealthMonitorServiceActor"
+
+        config {
+            papi-backend-name = PAPIv1
+            papi-v2-backend-name = PAPIv2
+
+            google-auth-name = service-account
+            gcs-bucket-to-check = "cromwell-ping-me-bucket"
```
[the remark this answers is not in the document]
Comment (cjllanwarne) [reply CHANGELOG.md]: There's more documentation in the [cromwell.examples.conf](https://github.com/broadinstitute/cromwell/blob/39/cromwell.example.backends/cromwell.examples.conf#L843) which I removed here for simplifying the instruction of "if your config looked like *that*, now instead make it look like *this*"

[the remark this answers is not in the document]
Comment (aednichols) [reply CHANGELOG.md]: That documentation doesn't answer my question either, but it does move this issue into ignorable "preexisting condition" territory relative to this PR

[the remark this answers is not in the document]
Comment (cjllanwarne) [reply CHANGELOG.md]: I'll see whether I can clarify that it's just a bucket name that we're asking for here (ie _not_ a URL and _not_ an object)

File: services/src/main/scala/cromwell/services/healthmonitor/impl/HealthMonitorServiceActor.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    val dockerHubSubsystemOption = if (serviceConfig.getBoolean("check-dockerhub")) Some(DockerHub) else None
+    val engineDatabaseSubsystemOption = if (serviceConfig.getBoolean("check-engine-database")) Some(EngineDb) else None
+    val gcsSubsystemOption = if (serviceConfig.getBoolean("check-gcs")) Some(Gcs) else None
+
+    (Set(
+      dockerHubSubsystemOption,
+      engineDatabaseSubsystemOption,
+      gcsSubsystemOption,
+    ) collect {
+      case Some(monitor) => monitor
```
Comment (mcovarr) [inline services/src/main/scala/cromwell/services/healthmonitor/impl/HealthMonitorServiceActor.scala]: I think `flatten`ing the `Set` would get you the same result

**Pull request merged**